### ISSUE TITLE
Propagate docker creation failure as containerStatus

### ISSUE THREAD

**Issue by dchen1107** [GoogleCloudPlatform/kubernetes#6321]
Kubelet current only populate create failure as events, not in ContainerStatus. We should propagate such failures through ContainerStatus. Before PodStatus is polling by master, and there is no way to obtain those failure except scanning docker.log. Now Kubelet directly post PodStatus, which makes capturing such failures much easier.

**Comment by yujuhong** [comment]
There are currently three defined container states.
```
type ContainerState struct {
     Waiting     *ContainerStateWaiting    `json:"waiting,omitempty"`
     Running     *ContainerStateRunning    `json:"running,omitempty"`
     Termination *ContainerStateTerminated `json:"termination,omitempty"`
 }
```
We can return "Waiting" with the reason field set to the creation failure. @dchen1107, does that sound good to you?

**Comment by dchen1107** [comment]
We currently do return Waiting, but without proper reason, instead of something like "Waiting for container run". We should propagate whatever error docker throw to us to upper layer, so that they can decide what to do next.

In a long run, I wish docker can incorporate those creation error to docker inspect result. I remembered there was an issue against docker, but couldn't find it now. Will file a new one.

**Comment by dchen1107** [comment]
My previous comment related to long run has problem. Yes, there is an issue against docker, but that is related to the failures cannot run docker container right after successful creation. When there is a creation failure, docker container object won't be created, and docker ps -a won't include such container in the list. 

Back to the original issue, now Kubelet post PodStatus directly, we can do something smarter to capture those failures without scanning docker.log

**Comment by dchen1107** [comment]
Fixed it via #6711
